compiler/ilgen/OMRMethodBuilder.cpp

```diff
@@ -620,7 +620,7 @@ OMR::MethodBuilder::getParameterTypes()
       TR_ASSERT_FATAL(symNamesIterator != _symbolNameFromSlot.end(), "No symbol found in slot %d", p);
       const char *name = symNamesIterator->second;
 
-      std::map<const char *, TR::IlType *, StrComparator>::iterator symTypesIterator = _symbolTypes.find(name);
+      auto symTypesIterator = _symbolTypes.find(name);
```

Comment:
For the record, I don't think the old code is necessarily "valid C++" as is suggested by the commit message. The type of `_symbolTypes` is `SymbolTypeMap`, which is a typedef for 

```c++
std::map<const char *, TR::IlType *, StrComparator, SymbolTypeMapAllocator>
```

Note how the custom allocator `SymbolTypeMapAllocator` is missing from:

```c++
std::map<const char *, TR::IlType *, StrComparator>::iterator symTypesIterator = _symbolTypes.find(name);
```

Not specifying the type of the allocator means it will be defaulted to the standard allocator. The different allocator types means the two `std::map` template instances are technically different types. So, the code is actually trying to assign an iterator from a container of one type to an interator of a container of a different type, which is undefined behaviour.

A possible reason the code works with the other compilers is that the allocator type isn't used in the definition of the iterator type in those STL implementations.

An alternative fix that should also work is to make `symTypesIterator`'s type be `SymbolTypeMap::iterator`. However, I am fine with using `auto` as is done in this change.